synapse/lib/stormlib/graph.py

```diff
@@ -0,0 +1,201 @@
+import synapse.exc as s_exc
+import synapse.lib.config as s_config
+import synapse.lib.stormtypes as s_stormtypes
+
+gdefSchema = {
+    'type': 'object',
+    'properties': {
+        'iden': {'type': 'string', 'pattern': s_config.re_iden},
+        'name': {'type': 'string', 'minLength': 1},
+        'scope': {'type': 'string', 'enum': ['user', 'global', 'power-up']},
+        'creatoriden': {'type': 'string', 'pattern': s_config.re_iden},
+        'creatorname': {'type': 'string', 'minLength': 1},
```

Comment:
lets not store user names in alternate locations

Reply:
I originally had this as just creator but split it up into `creatoriden` for user created graphs, and `creatorname` for power-up created graphs to keep them separate

Reply:
changed creatorname to power-up for clarity